### ISSUE TITLE
Is there an async version of GetRecords planned ?

### ISSUE THREAD

**Issue by TechWatching** [JoshClose/CsvHelper#1213]
I was wondering if there was of async version of GetRecords planned for the CsvReader. It seems odd that there are async versions for some methods and not this one. Am I missing something ?

**Comment by JoshClose** [comment]
There are only async methods where there needs to be one. The only async call that happens when reading is `TextReader.ReadAsync`, so only the methods that are up the chain from that are async.

If you want to run `GetRecords` on a separate thread/background worker/task, you should manage that on your side.

**Comment by TechWatching** [comment]
I saw that the Read method and a ReadAsync methods in CsvReader are using the CsvParser that himself uses the TextReader. So as the GetRecords method calls the Read method, I was thinking that there would have been a GetRecords async method that would have called the ReadAsync method. 

It's not really a need to me to have async code, the GetRecords method works fine for me. It's just that a GetRecordsAsync could have gain from the benefits of using await/async. But I just had a quick look to the code, perhaps I did not understand correctly.
